ClickHouse/ClickHouse
# Add `EXPLAIN TABLE OVERRIDE` for pre-validating overrides.

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `EXPLAIN TABLE OVERRIDE` query

Detailed description / Documentation draft:
The `EXPLAIN TABLE OVERRIDE` query can be used for pre-validating overrides. Example:

```sql
EXPLAIN TABLE OVERRIDE mysql('127.0.0.1:3306', 'db', 'table', 'user', 'pw')
    PARTITION BY tuple(toYYYYMM(created), id % 8)
```

Validations done:
 * check that storage overrides do not reference unknown or nullable columns
 * check that default specifier is not modified for columns
   (for example MATERIALIZED -> ALIAS)
